index.js

```diff
@@ -83,20 +83,30 @@ function runNgrok(opts, cb) {
 		return cb();
 	}
 
+	emitter.emit('statuschange', 'starting');
+
 	ngrok = spawn(
 			bin,
 			['start', '--none', '--log=stdout', '--region=' + opts.region, '-config=' + opts.configPath],
 			{cwd: __dirname + '/bin'});
 
-	ngrok.stdout.on('data', function (data) {
-		var addr = data.toString().match(ready);
-		if (addr) {
-			api = request.defaults({
-				baseUrl: 'http://' + addr[1],
-				json: true
-			});
-			cb();
-		}
+	ngrok.stdout.on('data', function (chunk) {
+		var lines = chunk.toString().split(/\r?\n/);
+		lines.forEach(function (line) {
+			if (!line) return;
+			var data = logfmt.parse(line);
+			if (data.obj === 'web' && data.msg === 'starting web service' && data.addr) {
+				api = request.defaults({
+					baseUrl: 'http://' + data.addr,
+					json: true
+				});
+				cb();
+			} else if (data.obj === 'csess' && data.msg === 'session closed, starting reconnect loop') {
```

Comment:
Should we maybe add tests to make sure that when we update ngrok's binary this doesn't break?

Reply:
Now I've added tests for the `statuschange` events with `starting` and `online` statuses. I don't know how to simulate the `session closed` scenario in the test, so I punted on adding a test for that. Do you maybe have an idea how that could work?

Reply:
I imagine we'd do something horrible like mess with ngrok's socket? I don't think we should worry about that too much though, having tests for a couple of the string parsing cases seems good enough to me!

Reply:
Messing with the socket sounded fun, so I had to give it a quick try 😄 Getting the socket file descriptor from `lsof` was easy enough and I was able to disrupt the connection by closing the socket manually using `lldb`. I wasn't able do this reliably from a script though.

I agree having couple of test cases for output parsing is good enough for now, so I'm going to merge this.